sherpa/astro/tests/test_astro_data2.py

```diff
@@ -1032,6 +1032,89 @@ def test_img_get_filter_compare_filtering(make_test_image):
     assert maska.max() == 1
 
 
+def test_pha_change_channels(make_test_pha):
+    """What happens if we change the channel/count values?
+
+    We have several ways of getting the independent and dependent
+    axes.
+    """
+    pha = make_test_pha
+
+    channels = [1, 2, 3, 4]
+    counts = [1, 2, 0, 3]
+    assert np.all(pha.channel == channels)
+    assert np.all(pha.counts == counts)
+
+    assert len(pha.get_indep()) == 1
+    assert np.all(pha.get_indep()[0] == channels)
+    assert np.all(pha.get_dep() == counts)
+
```

Comment:
Should we check `pha.x` and `pha.y`, too?

Reply:
Even though I try to catch everything, we at some point have to assume that "we have other tests that check, even if not under exactly the same condition" and hope we don't miss things (the coverage tests are useful but they unfortunately don't help us pick out certain paths that don't get covered).